### ISSUE TITLE
name property of ks.Series([1,2,3]) is '0' not None

### ISSUE THREAD

**Issue by charlesdong1991** [databricks/koalas#1010]
Just notice a Series behaviour in a WIP PR,

```python
>>> s1 = ks.Series([1,2,3])
>>> s1.name
'0'
```
is this designed on purpose? i think in pandas this should be `None`.

**Comment by HyukjinKwon** [comment]
I think it was because:

```python
>>> pd.DataFrame([1,2,3])
   0
0  1
1  2
2  3
```

However, for series, yes. I think we should fix. Maybe we should add an internal field name that's matched to `None` .. ?

cc @ueshin

**Comment by itholic** [comment]
closed since it's resolved in #1712 .
